src/pybamm/geometry/battery_geometry.py

```diff
@@ -30,6 +30,8 @@ def battery_geometry(
     """
     if options is None or type(options) == dict:  # noqa: E721
         options = pybamm.BatteryModelOptions(options)
+    if options["cell geometry"] == "cylindrical":
```

Comment:
Since `form_factor` is a kwarg of this method, I don't think we should update it in this way. Instead, we should catch it earlier and explicitly pass `form_factor` to `battery_geometry`